### ISSUE TITLE
Build streamlit dashboard for testing various algorithm/model parameters with mcts

### ISSUE THREAD

**Issue by nashirj** [ucfai/knightros-gambit#122]

**Comment by nashirj** [comment]
As part of this PR, we need to add [argparse](https://docs.python.org/3/library/argparse.html) to `train.py` and `stockfish_train.py` so that we can run the file with or without a dashboard. We'll use the following bash script to specify which mode of operation to use:
```bash
# example use:
# ./train.sh streamlit stockfish_train
# ./train.sh cli train

if [ "$2" = "train" ] || [ "$2" = "stockfish_train" ]; then
    if [ "$1" = "streamlit" ]; then
        streamlit run $2.py
    elif [ "$1" = "cli" ]; then
        python $2.py --cli
    else
        echo "Expected first argument to be either <streamlit> or <cli>"
    fi
else
    echo "Expected first argument to be either <train> or <stockfish_train>"
fi
```

In the `main` function, we'll check if the `--cli` flag was passed, and if so, read specified hyperparameters from a JSON file, otherwise, we'll be reading from the dashboard, and we'll set parameters accordingly.

**Comment by nashirj** [comment]
Closed in #167
